frontend/src/components/ui/AppDocumentUpload.vue

```diff
@@ -24,14 +35,25 @@
         </v-row>
       </div>
       <div v-if="uploadedDocuments.length > 0" class="mt-6 mx-4 mx-md-8 mx-lg-12">
-        <AppLabel>Uploaded Documents</AppLabel>
+        <AppLabel v-if="showTableHeader">Uploaded Documents</AppLabel>
         <v-data-table :headers="headersUploadedDocuments" :items="uploadedDocuments" item-key="documentId" items-per-page="-1" density="compact">
           <template #item.actionButtons="{ item }">
-            <v-icon v-if="!loading && !readonly" small @click="$emit('deleteUploadedDocument', item.documentId)">mdi-delete</v-icon>
+            <v-icon v-if="!loading && !readonly" small @click="$emit('deleteUploadedDocument', item.documentId, documentType)">mdi-delete</v-icon>
           </template>
           <template v-slot:bottom><!-- no paging --></template>
         </v-data-table>
       </div>
+      <div v-else-if="showRequiredMessage">
```

Comment:
Would it make more sense to accept the AppMissingInfoError.vue as a slot instead of repeating bits of here? It would reduce complexity in this file and AppMissingInfoError.vue. You wouldn't need showRequiredMessage.

Reply:
Fixed. Much better as a slot.